blog/2022/06-22-kubernetes-support/index.md

```diff
@@ -0,0 +1,275 @@
+
+Once this is done, Resoto will start collecting data from all configured clusters on the schedule configured in ResotoCore `resotocore.workflows.collect_and_cleanup.schedule`, which defaults to every hour. You can trigger a collection run manually by entering:
+
+```bash
+> workflows run collect_and_cleanup
+```
+
+## Kubernetes Resource Data Model
+
+Every resource found in Kubernetes is represented by a node in the graph. Available resource properties are captured and available to filter and automate.
+
+![PersistentVolume](./img/persistent_volume.svg)
+
+Resoto has a common data model for resources that abstracts from the underlying provider. Every resource inherits the properties from the base `resource`, which defines properties like `id`, `name`, `age`, `last_update`, `tags` etc. Every resource in Kubernetes is also of type `kubernetes_resource`, which brings in additional properties like a `resource_version`, `labels`, and an optional `namespace` property, which is defined for all namespaced resources. Annotations of a Kubernetes resource are available as `tags` and can be modified via the [`tag`](/docs/reference/cli/tag) command. Common abstractions apply as well. A [PersistentVolume](/docs/reference/data-models/kubernetes#kubernetes_persistent_volume) is of base type `volume`. So it shares the same kind and properties with other volumes of other cloud providers, e.g. an [`EBSVolume`](/docs/reference/data-models/aws#aws_ec2_volume) in AWS or a [`Disk`](/docs/reference/data-models/gcp#gcp_disk) in GCP. The same idea applies to a Kubernetes [node](/docs/reference/data-models/kubernetes#kubernetes_node) which is of base type `instance` etc.
+
+Kubernetes has its own way of describing a resource, which is available in Resoto as well. The three main sections `metadata`, `spec`, and `status` can be found in almost any resource. The data in the `metadata` section is basically covered by the base `resource` properties (`id`, `name`, `tags`, etc.). The `spec` section usually holds the desired state, while the `status` section holds the current state. Both sections are unique to the resource type. Since Resoto allows searching on any resource property, we made the `spec` and `status` section unique to the resource type. So a `Pod` in Resoto has a `pod_spec` and `pod_status` section, while a `Deployment` has a `deployment_spec` and `deployment_status` section. The relevant spec and status sections have the same specific data model as the Kubernetes resource.
+
+You can find a complete reference in our [resource reference](/docs/reference/data-models/kubernetes).
+
+## Searching the Graph
+
+Since all Kubernetes resources share the same base kind `kubernetes_resource`, we can filter the graph by this resource type. This is the output for the 2 clusters we have configured:
+
+```bash
+> search is(kubernetes_resource) | count kind
+
+kubernetes_job: 1
```

Comment:
I think a reader would benefit from understanding what they're looking at with the results. You describe the properties in the data model. I'd expect something like:

"Below is the output for the 2 clusters we have configured. You can see how the search result contains the different properties from Resoto's data model, and the corresponding value. "

something like that.